packages/block-editor/src/components/provider/index.js

```diff
@@ -138,9 +138,14 @@ export default compose( [
 			resetBlocks,
 		} = dispatch( 'core/block-editor' );
 
+		const {
+			getEditorSettings,
+		} = dispatch( 'core/editor' );
+
 		return {
 			updateSettings,
 			resetBlocks,
+			getEditorSettings,
```

Comment:
Actually, that's not exactly what I suggested. I thought we should add `codeEditingEnabled` to this list https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/provider/index.js#L76

and just use the existing `getSettings` selector.

Also, it would be good to document the new settings in https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/defaults.js#L28